JASP-Engine/JASP/R/ttestonesample.R

```diff
@@ -15,663 +15,443 @@
-					row.footnotes <- list(t = list(index))
-
-					row <- list(v = variable, df = "", p = "", m = "",
-								lowerCI = "", upperCI = "",
-								.footnotes = row.footnotes)
-					row[[testStat]] <- .clean(NaN)
-				}
-
-			## if we are not yet ready to perform, just create an empty table
-			} else {
-				row <- list(v = variable, df = ".", p = ".",
-							m = ".",  d = ".", lowerCI = ".", upperCI = ".")
-				row[[testStat]] <- "."
-			}
+.ttestOneSampleNormalTable <- function(jaspResults, dataset, options, ready, type) {
+  # Container
+  .ttestAssumptionCheckContainer(jaspResults, options, type)
+  container <- jaspResults[["AssumptionChecks"]]
+  if (!options$normalityTests || !is.null(container[["ttestNormalTable"]])) 
+    return()
+  container <- jaspResults[["AssumptionChecks"]]
+  # Create table
+  ttestNormalTable <- createJaspTable(title = "Test of Normality (Shapiro-Wilk)")
+  ttestNormalTable$showSpecifiedColumnsOnly <- TRUE
+  ttestNormalTable$position <- 2
+  
+  
+  ttestNormalTable$addColumnInfo(name = "v", title = "", type = "string")
+  ttestNormalTable$addColumnInfo(name = "W",   type = "number", title = "W")
+  ttestNormalTable$addColumnInfo(name = "p",   type = "pvalue", title = "p")
+  
+  message <- "Significant results suggest a deviation from normality."
+  ttestNormalTable$addFootnote(message)
+  
+  container[["ttestNormalTable"]] <- ttestNormalTable
+  
+  res <- try(.ttestOneSampleNormalFill(container, dataset, options, ready))
+  .ttestSetError(res, ttestNormalTable)
+}
 
+.ttestOneSampleMainFill <-function(jaspResults, dataset, options, ready, testStat, optionsList) {
+  direction <- switch(options$hypothesis,
+                   "notEqualToTestValue"  ="two.sided",
+                   "greaterThanTestValue" ="greater",
+                   "lessThanTestValue"    ="less")
+  
+  if(!ready)
+    return()
+  variables <- options$variables
+  if (length(variables) == 0)  
+    variables = "."
+  
+  rowNo      <- 1
+  ttest.rows <- list() # for each variable and each test, save stuff in there
+  
+  ## add a row for each variable, even before we are conducting tests
+  for (variable in variables)
+    ttest.rows[[length(ttest.rows) + 1]] <- list(v = variable)
+  
+  for (variable in variables) {
+    errors <- .hasErrors(dataset, 
+                         message = 'short', 
+                         type = c('observations', 'variance', 'infinity'),
+                         all.target = variable,
+                         observations.amount = '< 2')
+    for (test in seq_len(length(optionsList$whichTests))) {
+      currentTest <- optionsList$whichTests[[test]]
+      
+      ## don't run a test the user doesn't want
+      if (!currentTest) 
+        next
+      if (!identical(errors, FALSE)){
+        jaspResults[["ttest"]]$addFootnote(errors$message, colNames = c("p", "m", testStat), rowNames = variables)
+        row <- list(v = variable, p = "NaN", m = "NaN", lowerCI = "", upperCI = "")
```

Comment:
I'd just use the `testStat` column, to be consistent with the independent and paired samples ttests
(or do the other two like this)